Dockerfile

```diff
@@ -1,15 +1,28 @@
-# run with:
-# docker build -t lazydocker .
-# docker run -it lazydocker:latest /bin/sh -l
+ARG BASE_IMAGE_BUILDER=golang
+ARG BASE_IMAGE=alpine
+ARG ALPINE_VERSION=3.10
+ARG GO_VERSION=1.12.6
 
-FROM golang:alpine
+FROM ${BASE_IMAGE_BUILDER}:${GO_VERSION}-alpine${ALPINE_VERSION} AS builder
+ARG GOARCH=amd64
+ARG GOARM
+ARG VERSION=0.2.4
+RUN apk --update add -q --progress git
 WORKDIR /go/src/github.com/jesseduffield/lazydocker/
 COPY ./ .
-RUN CGO_ENABLED=0 GOOS=linux go build
+RUN CGO_ENABLED=0 GOOS=linux GOARCH=${GOARCH} GOARM=${GOARM} go build -a -installsuffix cgo -ldflags="-s -w \
+    -X main.commit=$(git rev-parse HEAD) \
+    -X main.version=${VERSION} \
+    -X main.buildSource=Docker"
 
-FROM alpine:latest
-RUN apk add -U git xdg-utils
-WORKDIR /go/src/github.com/jesseduffield/lazydocker/
-COPY --from=0 /go/src/github.com/jesseduffield/lazydocker /go/src/github.com/jesseduffield/lazydocker
-COPY --from=0 /go/src/github.com/jesseduffield/lazydocker/lazydocker /bin/
-RUN echo "alias gg=lazydocker" >> ~/.profile
+FROM ${BASE_IMAGE}:${ALPINE_VERSION}
+ARG VERSION=0.2.4
+LABEL org.label-schema.schema-version="1.0.0-rc1" \
+    maintainer="jessedduffield@gmail.com" \
+    org.label-schema.vcs-ref=$VCS_REF \
```

Comment:
should $VCS_REF be defined above?

Reply:
Oh yes I'll remove that line, that's for automated Docker builds really... By the way, would you have any idea why gocui fails with the error above? Because it would be nice to have the container run directly lazydocker as its entrypoint :)

Reply:
it's based on this:
```
func (g *Gui) SetView(name string, x0, y0, x1, y1 int, overlaps byte) (*View, error) {
	if x0 >= x1 {
		return nil, errors.New("invalid dimensions")
```

Seems like it's an issue with the window being too small. Perhaps the program starts before the terminal is attached. We might need to add some code in saying to wait until the window size is large enough before calling SetView

Reply:
I think If you want to run the container that’s need to  check and wait SIGWINCH signal

https://github.com/skanehira/docui/blob/08d7aa1ad7e5f417cffeba9fc2d5a627fda74221/common/common.go#L194

Reply:
That's nice. Since we are using `docker/docker` anyway, we can also import `github.com/docker/docker/pkg/term`, which provides the below `GetWinsize` function which returns the current dimensions of terminal, just like @skanehira does.

```go
	winSize, err := term.GetWinsize(os.Stdout.Fd())
	if err != nil {
		os.Exit(1)
	}

	if winSize.Width == 0 || winsize.Height == 0 {
		signalCh := make(chan os.Signal, 1)
		signal.Notify(signalCh, syscall.SIGWINCH, syscall.SIGINT)
		<-signalCh
	}
```

I haven't tested it though (waiting for go modules support in lazydocker), but I think it should be okay after minor tweaks. Just put this code before initializing Gui I guess.
Could someone test it @jesseduffield @qdm12 ? Or maybe someone has a better solution?

Reply:
There is a fix, using this script as entrypoint:

```sh
#!/bin/sh

resize > /dev/null
lazydocker $@
```

But it would indeed be cleaner to have it work in Go, so we could even have a scratch based container.